hub/api/dataset.py

```diff
@@ -16,6 +16,9 @@
 import warnings
 import os
 
+# Used to distinguish between attributes and items (tensors)
+DATASET_ATTRIBUTES = ["path", "mode", "slice", "provider", "tensors"]
```

Comment:
This needs to be established in the documentation as well. Can you please add inline documentation to the Dataset object where it is specified?